packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -521,38 +524,41 @@ Static group authorization should perform as expected.`,
   };
 
   private propagateAuthDirectivesToNestedTypes(type: ObjectTypeDefinitionNode, rules: AuthRule[], ctx: TransformerContext) {
+    let seenNonModelTypes: string[] = ctx.metadata.get(AUTH_NON_MODEL_TYPES);
```

Comment:
Const and I think using a Set here would be good to allow O(1) lookups which might matter for huge schemas

Also, it seems weird to me to have this method update the context in order to pass seen types up the call stack. I think it makes more sense for this to be an input to the method with a default value of [] and the recursive calls can pass in an updated set.

Better yet would be to get rid of the recursion entirely, push types onto a stack and drain the stack iteratively, but that doesn't have to happen in this PR

Reply:
Changed to a set.

If we did this are you referring to use a specific object for this as opposed to using the context metadata? We can get rid of the recursion though we'd still need a Set in the context metadata to check for seen types in between `@auth` executions.

Reply:
If it needs to be shared between separate executions, then it does make sense to put it in the context so that's fine.